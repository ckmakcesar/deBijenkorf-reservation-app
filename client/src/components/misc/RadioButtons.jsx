import React from 'react';
import clsx from 'clsx';
import PropTypes from 'prop-types';

import styles from '../../styles/misc/RadioButtons.module.css';

const RadioButtons = ({
  label,
  values,
  onChange,
  name,
  value: selectedValue,
  required,
  disabled,
  id,
}) => (
  <div className='input-field-label'>
    <div>{label ? `${label}${label && required ? '*' : ''}` : undefined}</div>

    {values.map((val, index) => (
      <div
        key={index}
        className={clsx(styles.radioButtonWithLabel, index > 0 && styles.paddingTop)}
      >
        <input
          {...(id ? { id: `Radio-button-${id}-${val}` } : null)} // optional id - should be unique
          type='radio'
          value={val}
          name={name}
          checked={selectedValue === val}
          onChange={e => { onChange(e.target.value); }}
          required={required}
          disabled={disabled}
        />
        <label
          htmlFor={val}
          onClick={() => { onChange(val); }}
        >
          {val}
        </label>
      </div>
    ))}
  </div>
);

// *********************************************************************************************************************
// PROPS
// *********************************************************************************************************************
RadioButtons.propTypes = {
  label: PropTypes.string,
  values: PropTypes.arrayOf(PropTypes.string).isRequired,
  onChange: PropTypes.func.isRequired,
  name: PropTypes.string,
  value: PropTypes.string,
  required: PropTypes.bool,
  disabled: PropTypes.bool,
  id: PropTypes.string,
};

RadioButtons.defaultProps = {
  label: '',
  name: '',
  value: '',
  required: false,
  disabled: false,
  id: '',
};

export default RadioButtons;
