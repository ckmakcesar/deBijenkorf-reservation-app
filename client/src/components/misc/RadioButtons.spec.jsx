import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

import RadioButtons from './RadioButtons';

describe('component:RadioButtons', () => {
  const mockGreetingChoices = ['Hola', 'Hello', 'Bonjour'];

  it('Should render RadioButtons with every optional props absent', () => {
    const wrapper = shallow(
      <RadioButtons
        values={mockGreetingChoices}
        onChange={jest.fn()}
      />,
    );
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });

  it('Should render RadioButtons with every optional props present', () => {
    const wrapper = shallow(
      <RadioButtons
        label='Group of Radio Buttons'
        values={mockGreetingChoices}
        onChange={jest.fn()}
        name='greetings-choices'
        value={mockGreetingChoices[1]}
        required={true}
        disabled={false}
      />,
    );
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});
