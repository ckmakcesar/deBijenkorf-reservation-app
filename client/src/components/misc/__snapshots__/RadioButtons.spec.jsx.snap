// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component:RadioButtons Should render RadioButtons with every optional props absent 1`] = `
<div
  className="input-field-label"
>
  <div />
  <div
    className="radioButtonWithLabel"
    key="0"
  >
    <input
      checked={false}
      disabled={false}
      name=""
      onChange={[Function]}
      required={false}
      type="radio"
      value="Hola"
    />
    <label
      htmlFor="Hola"
      onClick={[Function]}
    >
      Hola
    </label>
  </div>
  <div
    className="radioButtonWithLabel paddingTop"
    key="1"
  >
    <input
      checked={false}
      disabled={false}
      name=""
      onChange={[Function]}
      required={false}
      type="radio"
      value="Hello"
    />
    <label
      htmlFor="Hello"
      onClick={[Function]}
    >
      Hello
    </label>
  </div>
  <div
    className="radioButtonWithLabel paddingTop"
    key="2"
  >
    <input
      checked={false}
      disabled={false}
      name=""
      onChange={[Function]}
      required={false}
      type="radio"
      value="Bonjour"
    />
    <label
      htmlFor="Bonjour"
      onClick={[Function]}
    >
      Bonjour
    </label>
  </div>
</div>
`;

exports[`component:RadioButtons Should render RadioButtons with every optional props present 1`] = `
<div
  className="input-field-label"
>
  <div>
    Group of Radio Buttons*
  </div>
  <div
    className="radioButtonWithLabel"
    key="0"
  >
    <input
      checked={false}
      disabled={false}
      name="greetings-choices"
      onChange={[Function]}
      required={true}
      type="radio"
      value="Hola"
    />
    <label
      htmlFor="Hola"
      onClick={[Function]}
    >
      Hola
    </label>
  </div>
  <div
    className="radioButtonWithLabel paddingTop"
    key="1"
  >
    <input
      checked={true}
      disabled={false}
      name="greetings-choices"
      onChange={[Function]}
      required={true}
      type="radio"
      value="Hello"
    />
    <label
      htmlFor="Hello"
      onClick={[Function]}
    >
      Hello
    </label>
  </div>
  <div
    className="radioButtonWithLabel paddingTop"
    key="2"
  >
    <input
      checked={false}
      disabled={false}
      name="greetings-choices"
      onChange={[Function]}
      required={true}
      type="radio"
      value="Bonjour"
    />
    <label
      htmlFor="Bonjour"
      onClick={[Function]}
    >
      Bonjour
    </label>
  </div>
</div>
`;
